src/js/edu-benefits/pages/applicantInformation.js

```diff
@@ -59,6 +60,16 @@ export default function applicantInformation(schema, options) {
       'view:noSSN': {
         'ui:title': 'I don’t have a Social Security number',
       },
+      vaFileNumber: {
+        'ui:required': (formData) => !!_.get('view:noSSN', formData),
```

Comment:
Since this is such a common pattern, I think we really need a way to move as much logic out of the configs as possible and abstract it into its own file (which really means modifying `veteranId`'s `uiSchema` to allow for other field names). I have ideas for this...We could use default params and computed property names to allow for veteran vs relative information. Also passing in a `rootPath` object to preface the path in `ui:required` and `expandUnder`.

Should we do that here, or make another ticket for that?